lib/builder/util.go

```diff
@@ -63,15 +65,25 @@ func checkBuildEnv() error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	_, err = client.Version()
-	if err != nil {
-		log.Error(trace.DebugReport(err))
-		return trace.BadParameter(
-			`Docker does not seem to be available on this machine.
+	dockerErr := trace.BadParameter(
+		`Docker does not seem to be available on this machine.
 To resolve the issue:
   * Install Docker (https://docs.docker.com/engine/installation/).
   * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).`)
+
+	_, err = client.Version()
+	if err != nil {
+		log.Error(trace.DebugReport(err))
+		return dockerErr
+	}
+	_, err = client.Version()
+
+	err = utils.Exec(exec.Command("docker", "help"), ioutil.Discard)
+	if err != nil {
+		log.Error(err)
```

Comment:
I intentionally didn't used `trace.DebugReport` here because it was generating less useful output:

```
walt@gcp:~/git/gravity$ make -C assets/robotest images                                                                                                                                        
make: Entering directory '/home/walt/git/gravity/assets/robotest'                                                                                                                             
/home/walt/git/gravity/assets/robotest/tele_build.sh                                                                                                                                          + docker run --rm=true --net=host --user=1000:1000 --group-add=998 -v /home/walt/git/gravity/build/7.1.0-alpha.1.205/tele:/home/walt/git/gravity/build/7.1.0-alpha.1.205/tele -v /home/walt/gi
t/gravity/assets/robotest/current:/home/walt/git/gravity/assets/robotest/current -v /home/walt/git/gravity/build/7.1.0-alpha.1.205/packages:/home/walt/git/gravity/build/7.1.0-alpha.1.205/pac
kages -v /home/walt/git/gravity/build/7.1.0-alpha.1.205/robotest/tmp/tmp.HeHHFgZkMK:/home/walt/git/gravity/build/7.1.0-alpha.1.205/robotest/tmp/tmp.HeHHFgZkMK -v /var/run/docker.sock:/var/ru
n/docker.sock -v /run/docker.sock:/run/docker.sock --tmpfs /tmp -w /home/walt/git/gravity/build/7.1.0-alpha.1.205/robotest/tmp/tmp.HeHHFgZkMK quay.io/gravitational/debian-tall:buster-2020091
6 dumb-init /home/walt/git/gravity/build/7.1.0-alpha.1.205/tele build --debug /home/walt/git/gravity/assets/robotest/current/app.yaml --state-dir=/home/walt/git/gravity/build/7.1.0-alpha.1.2
05/packages --version=7.1.0-alpha.1.205 --output=/home/walt/git/gravity/build/7.1.0-alpha.1.205/robotest/tmp/tmp.HeHHFgZkMK/robotest-7.1.0-alpha.1.205.tar                                    ERRO [BUILDER]   "\nERROR REPORT:\nOriginal Error: *exec.Error exec: &#34;docker&#34;: executable file not found in $PATH\nStack Trace:\n\t/gopath/src/github.com/gravitational/gravity/lib/utils/exec.go:188 github.com/gravitational/gravity/lib/utils.ExecWithInput\n\t/gopath/src/github.com/gravitational/gravity/lib/utils/exec.go:182 github.com/gravitational/gravity/lib/utils.Exec\n\t/gopath/src/github.com/gravitational/gravity/lib/builder/util.go:81 github.com/gravitational/gravity/lib/builder.checkBuildEnv\n\t/gopath/src/github.com/gravitational/gravity/lib/builder/engine.go:120 github.com/gravitational/gravity/lib/builder.newEngine\n\t/gopath/src/github.com/gravitational/gravity/lib/builder/cluster.go:34 github.com/gravitational/gravity/lib/builder.NewClusterBuilder\n\t/gopath/src/github.com/gravitational/gravity/tool/tele/cli/build.go:75 github.com/gravitational/gravity/tool/tele/cli.buildClusterImage\n\t/gopath/src/github.com/gravitat
ional/gravity/tool/tele/cli/run.go:54 github.com/gravitational/gravity/tool/tele/cli.Run\n\t/gopath/src/github.com/gravitational/gravity/tool/tele/main.go:44 main.run\n\t/gopath/src/github.c
om/gravitational/gravity/tool/tele/main.go:35 main.main\n\t/go/src/runtime/proc.go:203 runtime.main\n\t/go/src/runtime/asm_amd64.s:1357 runtime.goexit\nUser Message: exec: &#34;docker&#34;: 
executable file not found in $PATH" builder/util.go:83                                                                                                                                        
ERRO             Command failed. error:[                                                                                                                                                      
ERROR REPORT:                                                                                                                                                                                 
Original Error: *trace.BadParameterError Docker does not seem to be available on this machine.                                                                                                
To resolve the issue:                                                                                                                                                                         
  * Install Docker (https://docs.docker.com/engine/installation/).                                                                                                                            
  * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).                                                                                   
Stack Trace:                                                                                                                                                                                  
        /gopath/src/github.com/gravitational/gravity/lib/builder/util.go:68 github.com/gravitational/gravity/lib/builder.checkBuildEnv                                                                /gopath/src/github.com/gravitational/gravity/lib/builder/engine.go:120 github.com/gravitational/gravity/lib/builder.newEngine                                                         
        /gopath/src/github.com/gravitational/gravity/lib/builder/cluster.go:34 github.com/gravitational/gravity/lib/builder.NewClusterBuilder                                                 
        /gopath/src/github.com/gravitational/gravity/tool/tele/cli/build.go:75 github.com/gravitational/gravity/tool/tele/cli.buildClusterImage                                               
        /gopath/src/github.com/gravitational/gravity/tool/tele/cli/run.go:54 github.com/gravitational/gravity/tool/tele/cli.Run                                                               
        /gopath/src/github.com/gravitational/gravity/tool/tele/main.go:44 main.run                                                                                                            
        /gopath/src/github.com/gravitational/gravity/tool/tele/main.go:35 main.main                                                                                                                   /go/src/runtime/proc.go:203 runtime.main                                                                                                                                                      /go/src/runtime/asm_amd64.s:1357 runtime.goexit                                                                                                                                       User Message: Docker does not seem to be available on this machine.                                                                                                                           To resolve the issue:                                                                                                                                                                           * Install Docker (https://docs.docker.com/engine/installation/).                                                                                                                            
  * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).] tele/main.go:36                                                                  
[ERROR]: Docker does not seem to be available on this machine.                                                                                                                                
To resolve the issue:                                                                                                                                                                         
  * Install Docker (https://docs.docker.com/engine/installation/).                                                                                                                            
  * Make sure it can be used by a non-root user (https://docs.docker.com/install/linux/linux-postinstall/).                                                                                   
make: *** [Makefile:201: /home/walt/git/gravity/build/7.1.0-alpha.1.205/robotest/images/robotest-7.1.0-alpha.1.205.tar] Error 255                                                             
make: Leaving directory '/home/walt/git/gravity/assets/robotest'    
```

Reply:
In general with debugging gravity errors, the stacks tend to be quite important. The example you used stands on it's own, but most of our errors aren't currently useful without the added context of the path used to reach the function. One of those hidden tricks is to get used to typing: `| sed 's/\\n/\n/g' | sed 's/\\t/\t/g'` to format the output when viewing.

Reply:
These messages were reworked in 2ae2d38  per Roman's suggestion here: 
https://github.com/gravitational/gravity/pull/2150#discussion_r494706145